tools/manually-handling-sharding-ddl-locks.md

```diff
@@ -0,0 +1,415 @@
+
+The sharding DDL lock of TiDB-DM automatically completes the synchronization process in most cases, but under some abnormal scenarios you need the command `unlock-ddl-lock` or `break-ddl-lock` to handle abnormal DDL locks manually.
+
+> **Warning:**
+>- Do not use `unlock-ddl-lock` or `break-ddl-lock` unless you are totally aware of the possible impacts brought by the command and you can accept them.
+>- Before manually handling the abnormal DDL lock, make sure that you have already read the DM [shard merge principles](./shard-merge.md#Principles).
+
+### Command
+
+#### show-ddl-locks
+
+Show the current DDL lock information on `DM-master`.
+
+##### Command usage
+
+```bash
+show-ddl-locks [--worker=127.0.0.1:8262] [task-name]
+```
+
+##### Variables description
+
+- `worker`: flag; string; `--worker`; optional; can be specified multiple times; if not being specified, show the lock information of all DM-workers; if being specified, show the lock information of the specified DM-worker.
+- `task-name`: non-flag; string; optional; if not being specified, show the lock information of all tasks; if being specified, show the lock information of the specified task.
+
+##### Example of results
+
+```bash
+» show-ddl-locks test
+{
+    "result": true,                                        # show if the locking process succeeds
+    "msg": "",                                             # show the reason for the locking process failure or other descriptive information (for example, the locking task does not exist)
+    "locks": [                                             # the lock information list on the DM-master
+        {
+            "ID": "test-`shard_db`.`shard_table`",         # the ID of the lock, which is composed of the current task name and the corresponding schema/table information of the DDL
+            "task": "test",                                # the task name of the lock
+            "owner": "127.0.0.1:8262",                     # the owner of the lock (the first DM-worker which receives the DDL event)
+            "DDLs": [                                      # the corresponding DDL list of the lock
+                "USE `shard_db`; ALTER TABLE `shard_db`.`shard_table` DROP COLUMN `c2`;"
+            ],
+            "synced": [                                    # the list of DM-workers that have received all sharding DDL events in the corresponding MySQL instances
+                "127.0.0.1:8262"
+            ],
+            "unsynced": [                                  # the list of DM-workers that have not yet received all sharding DDL events in the corresponding MySQL instances
+                "127.0.0.1:8263"
+            ]
+        }
+    ]
+}
+```
+
+#### unlock-ddl-lock
+
+Actively ask `DM-master` to unlock the specified DDL lock, for example, ask the owner to execute the DDL; ask other DM-workers that are not the owner to execute or skip the DDL; remove the lock information on `DM-master`.
```

Comment:
1. maybe `including` is better than `for example` here.
2. `ask other DM-workers that are not the owner to skip the DDL`